plugins/collection/class.tx_dlf_collection.php

```diff
@@ -154,135 +134,130 @@ protected function showCollectionList() {
+        $parameters = array ("fl" => "uid", "sort" => "uid asc");
 
-            while ($resArrayVolumes = $GLOBALS['TYPO3_DB']->sql_fetch_assoc($resultVolumes)) {
+        // Process results.
+        foreach ($collections as $collection) {
 
-                $volumes[$resArrayVolumes['uid']] = $resArrayVolumes['volumes'];
+            $solr_query = '';
 
+            if ($collection['index_query'] != "") {
+                $solr_query .= '('.$collection['index_query'].')';
+            } else {
+                $solr_query .= 'collection:'.'"'.$collection['index_name'].'"';
             }
 
-            // Process results.
-            while ($resArray = $GLOBALS['TYPO3_DB']->sql_fetch_assoc($result)) {
-
-                // Generate random but unique array key taking priority into account.
-                do {
-
-                    $_key = ($resArray['priority'] * 1000) + mt_rand(0, 1000);
-
-                } while (!empty($markerArray[$_key]));
-
-                // Merge plugin variables with new set of values.
-                $additionalParams = array ('collection' => $resArray['uid']);
-
-                if (is_array($this->piVars)) {
-
-                    $piVars = $this->piVars;
+            $partOfNothing = $solr->search_raw($solr_query.' partof:0', $parameters);
+            $partOfSomething = $solr->search_raw($solr_query.' AND NOT partof:0', $parameters);
 
-                    unset($piVars['DATA']);
+            $titles = array ();
 
-                    $additionalParams = tx_dlf_helper::array_merge_recursive_overrule($piVars, $additionalParams);
-
-                }
-
-                // Build typolink configuration array.
-                $conf = array (
-                    'useCacheHash' => 1,
-                    'parameter' => $GLOBALS['TSFE']->id,
-                    'additionalParams' => \TYPO3\CMS\Core\Utility\GeneralUtility::implodeArrayForUrl($this->prefixId, $additionalParams, '', TRUE, FALSE)
-                );
-
-                // Link collection's title to list view.
-                $markerArray[$_key]['###TITLE###'] = $this->cObj->typoLink(htmlspecialchars($resArray['label']), $conf);
-
-                // Add feed link if applicable.
-                if (!empty($this->conf['targetFeed'])) {
-
-                    $img = '<img src="'.\TYPO3\CMS\Core\Utility\ExtensionManagementUtility::siteRelPath($this->extKey).'res/icons/txdlffeeds.png" alt="'.$this->pi_getLL('feedAlt', '', TRUE).'" title="'.$this->pi_getLL('feedTitle', '', TRUE).'" />';
+            foreach ($partOfNothing as $doc) {
+                $titles[] = $doc->uid;
+            }
 
-                    $markerArray[$_key]['###FEED###'] = $this->pi_linkTP($img, array ($this->prefixId => array ('collection' => $resArray['uid'])), FALSE, $this->conf['targetFeed']);
+            $volumes = $titles;
 
-                } else {
+            // All titles except those that are part of something
```

Comment:
You confused titles and volumes here. Titles are all documents which do not have an ancestor (i.e. partof == 0), but volumes are all documents which either have an ancestor (i.e. partof != 0) or no descendant (i.e. no other documents exist with descendant.partof == ancestor.uid).
The way you calculate titles and volumes always results in less volumes than titles, but in reality that can never be the case. The number of volumes must be at least as big as the number of titles, almost always it is even bigger.